src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,181 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.setPrev(prev);
+            this.setNext(next);
+            this.setValue(value);
+        }
+
+        public Node<T> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<T> prev) {
+            this.prev = prev;
+        }
+
+        public Node<T> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<T> next) {
+            this.next = next;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public void setValue(T value) {
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+            tail = head;
+        } else {
+            tail.setNext(newNode);
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            if (head == null) {
+                add(value);
+            } else {
+                Node<T> newNode = new Node<>(null, value, head);
+                head.setPrev(newNode);
+                head = newNode;
+                size++;
+            }
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nodeOnIndex = find(index);
+        Node<T> newNode = new Node<>(nodeOnIndex.getPrev(), value, nodeOnIndex);
+        nodeOnIndex.getPrev().setNext(newNode);
+        nodeOnIndex.setPrev(newNode);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        list.forEach(this::add);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return find(index).getValue();
+    }
+
+    private Node<T> find(int index) {
```

Comment:
Read about [Ordering of class contents](https://mate-academy.github.io/style-guides/java/java.html#s3.4.2-ordering-class-contents)